src/ast/ddl.rs

```diff
@@ -2002,22 +2002,45 @@ impl fmt::Display for DropBehavior {
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
 pub enum UserDefinedTypeRepresentation {
+    /// Composite type: `CREATE TYPE name AS (attributes)`
     Composite {
         attributes: Vec<UserDefinedTypeCompositeAttributeDef>,
     },
+    /// Enum type: `CREATE TYPE name AS ENUM (labels)`
+    ///
     /// Note: this is PostgreSQL-specific. See <https://www.postgresql.org/docs/current/sql-createtype.html>
     Enum { labels: Vec<Ident> },
+    /// Range type: `CREATE TYPE name AS RANGE (options)`
+    Range {
+        options: Vec<UserDefinedTypeRangeOption>,
+    },
+    /// Base type (SQL definition): `CREATE TYPE name (options)`
+    ///
+    /// Note the lack of `AS` keyword
+    SqlDefinition {
+        options: Vec<UserDefinedTypeSqlDefinitionOption>,
+    },
```

Comment:
Could we include a link to the postgres docs for the new variants?

Reply:
I wanted to, but they do not have a linkable subsection so the link would be to the same `CREATE TYPE` page. The sub-headers do not have the anchor link unfortunately.

Reply:
Ok, doing that.

Reply:
Added